src/Models/OrderModel.php

```diff
@@ -93,11 +113,57 @@ public function addOrder(OrderEntityInterface $orderEntity): bool
         return true;
     }
 
+    /**
+     * Cancels an order in the Database through the following transaction:
+     *  - soft deletes order in the orders table
+     *  - updates products' stockLevels with old stockLevel plus the relevant volumeOrdered from order.
+     *
+     * @param string $orderNumber
+     * @return bool depending on whether the transaction was successful or not.
+     */
+    public function cancelOrder(string $orderNumber) {
+        $this->db->beginTransaction();
+
+        $cancelOrderQuery = $this->db->prepare("UPDATE `orders`
+                                                    SET `deleted` = 1
+                                                    WHERE `orderNumber` = ?");
+        $cancelOrderQueryResult = $cancelOrderQuery->execute([$orderNumber]);
+
+        if (!$cancelOrderQueryResult) {
+            $this->db->rollback();
+            return false;
+        }
+
+        $productsOrdered = $this->getOrderedProductsByOrderNumber($orderNumber);
+
+        if (!$productsOrdered) {
```

Comment:
```suggestion
        if ($productsOrdered === false) {
```
I think otherwise if there are no ordered products the transaction will fail.